### PR TITLE
Auto-focus in search box on search page

### PR DESCRIPTION
**Pull request by mrvisser**
When the search page is loaded, focus should automatically be set on the search box. Note that the cursor should be set at the end of the text in the field.

### CODE REVIEW

**Comment by mrvisser** [comment]
Looks like the `oae.api.js` initialization process can wreck autofocus, so I just restore it after initial widgets are loaded

**Comment by mrvisser** [comment]
Assigned to @nicolaasmatthijs for review

File: shared/oae/api/oae.api.js [hunk 1]
```diff
@@ -132,6 +132,11 @@ define(['oae.api.admin', 'oae.api.authentication', 'oae.api.config', 'oae.api.co
                                         // initial widget loading have finished
                                         $('body').css('visibility', 'visible');
 
+                                        // Apply auto-focus after the core HTML widgets have loaded.
```
Comment (nicolaasmatthijs) [inline shared/oae/api/oae.api.js]: Neat solution

File: shared/oae/api/oae.api.js [hunk 2]
```diff
@@ -132,6 +132,11 @@ define(['oae.api.admin', 'oae.api.authentication', 'oae.api.config', 'oae.api.co
...
+                                        // The rendering will cause HTML5 autofocus to be lost, so
+                                        // we re-inforce it here
+                                        $('[autofocus]:first').focus();
```
Comment (nicolaasmatthijs) [inline shared/oae/api/oae.api.js]: I always get the cursor at the beginning of the text field. The cursor should instead be positioned at the end of the value of the text field.

**Comment by nicolaasmatthijs** [comment]
Finished review. Back to @mrvisser for follow-up

**Comment by mrvisser** [comment]
It was intermittent. If the value gets set in search.js before the autofocus happens, then the focus would put the cursor to the beginning.

**Comment by mrvisser** [comment]
Back to @nicolaasmatthijs

**Comment by nicolaasmatthijs** [comment]
Looks good. Merged.

**Pull request merged**